messaging/src/test/java/org/axonframework/messaging/MessageTypeTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging;
+
+import org.junit.jupiter.api.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test class validating the {@link MessageType}.
+ *
+ * @author Steven van Beelen
+ */
+class MessageTypeTest {
+
+    private static final String NAMESPACE = MessageTypeTest.class.getPackageName();
+    private static final String LOCAL_NAME = MessageTypeTest.class.getSimpleName();
+    private static final String NAME = NAMESPACE + "." + LOCAL_NAME;
+    private static final QualifiedName QUALIFIED_NAME = new QualifiedName(NAME);
+    private static final String VERSION = "5.0.0";
```

Comment:
Please check what if name or version includes `#` (version delimiter) if everything works or we need to do handle that somehow.